zellij-utils/src/input/mouse.rs

```diff
@@ -2,102 +2,146 @@ use serde::{Deserialize, Serialize};
 
 use crate::position::Position;
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Deserialize, Serialize)]
+/// A mouse event can have any number of buttons (including no
+/// buttons) pressed or released.
+pub struct MouseEvent {
+    /// A mouse event can current be a Press, Release, or Motion.
+    /// Future events could consider double-click and triple-click.
+    pub event_type: MouseEventType,
+
+    // Mouse buttons associated with this event.
+    pub left: bool,
+    pub right: bool,
+    pub middle: bool,
+    pub wheel_up: bool,
+    pub wheel_down: bool,
+
+    // Keyboard modifier flags can be encoded with events too.  They
+    // are not often passed on the wire (instead used for
+    // selection/copy-paste and changing terminal properties
+    // on-the-fly at the user-facing terminal), but alt-mouseclick
+    // usually passes through and is testable on vttest.  termwiz
+    // already exposes them too.
+    pub shift: bool,
+    pub alt: bool,
+    pub ctrl: bool,
+
```

Comment:
Also FYI that Swing and WIN32 both expose keyboard modifiers with mouse events.  Soooo when zellij is running natively on Windows we'll have the infrastructure to do things like Ctrl-Click / Alt-Drag / etc. :-)

Reply:
Honestly, even `Alt` + `mouse_click` would be amazing. I'm thinking of using it for multiple selection of panes.